harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -166,12 +166,9 @@ def download(self, path: Optional[str] = None, mode: DownloadMode = DownloadMode
             mode (DownloadMode, optional): Governs how a checkpoint is downloaded. Defaults to
                 ``AUTO``.
         """
-        self.reload()
-        assert self.state
-        if (
-            self.state != CheckpointState.COMPLETED
-            and self.state != CheckpointState.PARTIALLY_DELETED
-        ):
+        if self.state not in [CheckpointState.COMPLETED, CheckpointState.PARTIALLY_DELETED]:
+            if self.state is None:
+                raise ValueError("Checkpoint state is unknown. Please reload the checkpoint.")
```

Comment:
maybe be more explicit with the language here? like, "call reload() on the Checkpoint" or something? on first glance, i don't know what it means to reload the checkpoint.